.github/workflows/ci.yaml

```diff
@@ -0,0 +1,52 @@
+on:
+  push:
+    branches:
+      - master, github-action
+  pull_request:
+
+name: Gridiron
+
+jobs:
+  check:
+    name: Check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: check
+
+  test:
```

Comment:
Why run three separate jobs instead of one? Wouldn't it be faster to run a single job?

Asking the question a different way: Is there a time when you'd want to run one of these jobs without the other two?

Reply:
Good question. I'm just following the recipe from https://github.com/actions-rs/meta/blob/master/recipes/quickstart.md

Do they split it into 3 jobs so they can all run in parallel? Maybe it's for better reporting? :man_shrugging:

Reply:
I was wrong, I didn't know this was a thing. I started with the default rust action.

This does allow them to be done parallel, which will speed up the checks but will use more minutes. I think that if one fails the others will still execute. In the TSP I used one job with steps so that I could do the fast checks first to not run the slow ones if they fail.

They also aren't caching build artifacts as far as I can tell in those actions, which is a big deal for the TSP, and is where most of the noise in its workflow config comes from.